src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -56,30 +37,76 @@ class WorkflowActor extends Actor {
         requiredInput._1 -> s"Incompatible workflow input types, expected $expected, got ${requiredInput._2}"
     }
 
-  override def receive: Receive = {
+    if (diagnostics.nonEmpty) {
+      throw new UnsatisfiedInputsException(diagnostics)
+    }
 
-    case Construct(binding, actualInputs) if !_constructed =>
-      val unsatisfiedInputs = checkUnsatisfiedInputs(binding.workflow, actualInputs)
-      if (unsatisfiedInputs.isEmpty) {
-        _constructed = true
-        sender ! Constructed
-      } else {
-        sender ! ConstructionFailed(unsatisfiedInputs)
-      }
-
-    case msg if !_constructed =>
-      sender ! InvalidOperation(s"Received $msg but WorkflowActor not Constructed!")
-
-    case msg@Construct(_, _) if _constructed =>
-      sender ! InvalidOperation(s"Received $msg but WorkflowActor already Constructed!")
-
-    case Start =>
-      sender ! Started
-
-    case Stop =>
-      // TODO is this the right way to stop this actor?
-      // TODO http://stackoverflow.com/questions/13847963/akka-kill-vs-stop-vs-poison-pill
-      context.stop(self)
-      sender ! Stopped
+    Props(new WorkflowActor(binding, actualInputs))
+  }
+}
+
+/** Represents the root of a single workflow instance, not a manager of multiple
+  * workflows.
+  */
+class WorkflowActor private(binding: WdlBinding, actualInputs: Map[FullyQualifiedName, WdlValue]) extends Actor {
+
+  import WorkflowActor._
+
+  // That which started the workflow, a role which has not yet been defined more specifically.
+  private var _primeMover: ActorRef = _
+
+  private def isStarted: Boolean = _primeMover != null
+
+  override def receive: Receive = {
+    LoggingReceive {
+
+      case Start =>
+        _primeMover = sender()
+        val executionTable = new ExecutionTable(binding)
+        val symbolTable = new SymbolTable(binding, actualInputs)
+        executionTable.runnableCalls.foreach { call =>
```

Comment:
I feel that if one finds themselves making a multi-line foreach/map then one should consider moving that code into a function and making it a one line call